api/v1alpha1/api.go

```diff
@@ -106,8 +106,30 @@ type LLMBackendSpec struct {
 	//
 	// +kubebuilder:validation:Required
 	BackendRef egv1a1.BackendRef `json:"backendRef"`
+
+	// BackendSecurityPolicyName list the BackendSecurityPolicy that this backend will depend on.
+	//
+	// A BackendSecurityPolicy is for configuring authentication and authorization rules on the traffic
+	// exiting the gateway to the backend.
+	//
+	// +optional
+	BackendSecurityPolicyName *string `json:"backendSecurityPolicyName,omitempty"`
+
+	// LLMProviderType is the provider type specifies the provider associated with the backend.
+	//
+	// +kubebuilder:validation:Enum=AWS;Unspecified
+	// +kubebuilder:default=Unspecified
+	ProviderType *LLMProviderType `json:"providerType"`
```

Comment:
Do we need this again? we can derive provider type from outputSchema(LLMAPISchema--> APISchema)

Reply:
I wanted to keep them distinct because I didn't think the schemas are uniquely supported by each provider. Open to changing it if it's guaranteed schema == provider

Reply:
I think schema is provider specific atm, not sure will it change in future. @mathetake 
Also currently APISchema is string type which allow constant such as "OpenAI, AwsBedrock" It may redundant to keep both field.

Reply:
There are cases that LLM provider adopts the OpenAI schema, so the provider name and APISchema name may not be necessarily the same. The provider determines the authentication mechanism and APISchema.